dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/Utils.java

```diff
@@ -917,33 +931,48 @@ public Object getDSpaceAPIObjectFromRest(Context context, BaseObjectRest restObj
     public BaseObjectRest getBaseObjectRestFromUri(Context context, String uri) throws SQLException {
         String dspaceUrl = configurationService.getProperty("dspace.server.url");
         // first check if the uri could be valid
-        if (!StringUtils.startsWith(uri, dspaceUrl)) {
-            throw new IllegalArgumentException("the supplied uri is not valid: " + uri);
+        if (!urlIsPrefixOf(dspaceUrl, uri)) {
+            throw new IllegalArgumentException("the supplied uri is not ours: " + uri);
+        }
+
+        // Extract from the URI the category, model and id components.
+        // They start after the dspaceUrl/api/{apiCategory}/{apiModel}/{id}
+        URL dspaceUrlObject;
+        URL requestUrlObject;
+        try {
+            dspaceUrlObject = new URL(dspaceUrl);
+            requestUrlObject = new URL(uri);
+        } catch (MalformedURLException ex) {
```

Comment:
we should think if this need to go before the prefix check. Eventually, the reason for that is that if the supplied uri is not a syntactically valid URI a BadRequest exception could be more appropriate than a potential 422 error reported if the URI is not valid or lead to a not appropriate object

Reply:
A good suggestion.  Done.